mathesar/tests/integration/test_ui.py

```diff
@@ -0,0 +1,72 @@
+import re
+from playwright.sync_api import Page, expect, Locator
+from mathesar.tests.integration.utils.locators import get_table_entry
+
+
+def create_new_empty_table(page: Page):
+    page.click("[aria-label='New Table']")
+    page.click("button:has-text('Empty Table')")
+
+
+def get_tab(page: Page, tab_text):
+    return page.locator(
+        f".tab-container [role=tablist] [role=presentation]:has-text('{tab_text}')"
+    )
+
+
+def close_tab(tab_locator: Locator):
+    tab_locator.hover()
+    tab_locator.locator("[aria-label=remove]").click()
+
+
+def test_tabs(page: Page, base_schema_url):
+    page.goto(base_schema_url)
+
+    # Create Table 0
+    create_new_empty_table(page)
+    table_0_entry = get_table_entry(page, "Table 0")
+    table_0_tab = get_tab(page, "Table 0")
+    close_tab(table_0_tab)
+
+    # Create Table 1
+    create_new_empty_table(page)
+    table_1_entry = get_table_entry(page, "Table 1")
+    table_1_tab = get_tab(page, "Table 1")
+    close_tab(table_1_tab)
+
+    # No tabs should be open
+    expect(table_0_tab).not_to_be_visible()
+    expect(table_1_tab).not_to_be_visible()
+
+    # Open Table 0
+    table_0_entry.click()
+    expect(table_0_entry.locator(".item")).to_have_class(re.compile("active"))
```

Comment:
> Do we need to use a regex

Unfortunately, yes -- as far as I can tell. I initially tried a plain string and it did not work. The [`to_have_class` locator assertion](https://playwright.dev/python/docs/1.18/api/class-locatorassertions#locator-assertions-to-have-class) compares the expected value to the _entire_ value of the `class` attribute. So if you have `<div class="foo bar" />` a call of `.to_have_class("foo")` will fail because `"foo" != "foo bar"`. I found this behavior surprising. It also made me wonder whether